contracts/staking/test/unit_tests/mixin_cumulative_rewards_test.ts

```diff
@@ -0,0 +1,258 @@
+            const epochOfFirstReward = new BigNumber(1);
+            const epochOfSecondReward = new BigNumber(2);
+            const epochOfIntervalStart = new BigNumber(1);
+            const epochOfIntervalEnd = new BigNumber(3);
+            await runTest(
+                amountToStake,
+                epochOfFirstReward,
+                epochOfSecondReward,
+                epochOfIntervalStart,
+                epochOfIntervalEnd,
+            );
+        });
+
+        it('Should successfully compute reward if no entry for current epoch, but there is an entry for epoch n-2', async () => {
+            // End epoch = n-2 forces the code to query the most recent cumulative reward
+            const amountToStake = toBaseUnitAmount(1);
+            const epochOfFirstReward = new BigNumber(1);
+            const epochOfSecondReward = new BigNumber(2);
+            const epochOfIntervalStart = new BigNumber(1);
+            const epochOfIntervalEnd = new BigNumber(4);
+            await runTest(
+                amountToStake,
+                epochOfFirstReward,
+                epochOfSecondReward,
+                epochOfIntervalStart,
+                epochOfIntervalEnd,
+            );
+        });
+
+        it('Should successfully compute reward are no cumulative reward entries', async () => {
+            // No entries forces the default cumulatie reward to be used in computations
+            const stake = toBaseUnitAmount(1);
+            const beginEpoch = new BigNumber(1);
+            const endEpoch = new BigNumber(2);
+            const reward = await testContract.computeMemberRewardOverInterval.callAsync(
+                testPoolId,
+                stake,
+                beginEpoch,
+                endEpoch,
+            );
+            expect(reward).to.bignumber.equal(ZERO);
+        });
+
+        it('Should return zero if no stake was delegated', async () => {
+            const stake = toBaseUnitAmount(0);
+            const beginEpoch = new BigNumber(1);
+            const endEpoch = new BigNumber(2);
+            const reward = await testContract.computeMemberRewardOverInterval.callAsync(
+                testPoolId,
+                stake,
+                beginEpoch,
+                endEpoch,
+            );
+            expect(reward).to.bignumber.equal(ZERO);
+        });
+
+        it('Should return zero if the start/end of the interval are the same epoch', async () => {
+            const stake = toBaseUnitAmount(1);
+            const beginEpoch = new BigNumber(1);
+            const endEpoch = new BigNumber(1);
+            const reward = await testContract.computeMemberRewardOverInterval.callAsync(
+                testPoolId,
+                stake,
+                beginEpoch,
+                endEpoch,
+            );
+            expect(reward).to.bignumber.equal(ZERO);
+        });
+
+        it('Should revert if start is greater than the end of the interval', async () => {
```

Comment:
Despite the fact that this should never happen, I think it could make sense to test for the case when we don't have the right "endpoints." Basically, when we are missing the cumulative reward values that we need to calculate rewards.

Reply:
This is handled by `Should successfully compute reward are no cumulative reward entries` - in this case the default CR is used.